kafka-client/src/test/java/dev/responsive/kafka/internal/db/mongo/MongoTelemetryListenerTest.java

```diff
@@ -0,0 +1,164 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.mongo;
+
+import static dev.responsive.kafka.internal.db.mongo.MongoTelemetryListener.commandsFailedCount;
+import static dev.responsive.kafka.internal.db.mongo.MongoTelemetryListener.commandsFailedLatency;
+import static dev.responsive.kafka.internal.db.mongo.MongoTelemetryListener.commandsSucceededCount;
+import static dev.responsive.kafka.internal.db.mongo.MongoTelemetryListener.commandsSucceededLatency;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.mongodb.event.CommandFailedEvent;
+import com.mongodb.event.CommandSucceededEvent;
+import dev.responsive.kafka.internal.metrics.ResponsiveMetrics;
+import dev.responsive.kafka.internal.metrics.exporter.MetricsExportService;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.Metrics;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class MongoTelemetryListenerTest {
+
+  @Test
+  public void testSuccessMetricNames() {
+    final MetricName findSuccessCountName = commandsSucceededCount("find");
+    assertEquals("responsive-mongodb", findSuccessCountName.group());
+    assertEquals("commands-succeeded-count", findSuccessCountName.name());
+    assertEquals(Collections.singletonMap("command", "find"), findSuccessCountName.tags());
+
+    final MetricName findSuccessLatencyName = commandsSucceededLatency("find");
+    assertEquals("responsive-mongodb", findSuccessLatencyName.group());
+    assertEquals("commands-succeeded-cumulative-latency", findSuccessLatencyName.name());
+    assertEquals(Collections.singletonMap("command", "find"), findSuccessLatencyName.tags());
+  }
+
+  @Test
+  public void testFailureMetricNames() {
+    final MetricName findFailureCountName = commandsFailedCount("find");
+    assertEquals("responsive-mongodb", findFailureCountName.group());
+    assertEquals("commands-failed-count", findFailureCountName.name());
+    assertEquals(Collections.singletonMap("command", "find"), findFailureCountName.tags());
+
+    final MetricName findFailureLatencyName = commandsFailedLatency("find");
+    assertEquals("responsive-mongodb", findFailureLatencyName.group());
+    assertEquals("commands-failed-cumulative-latency", findFailureLatencyName.name());
+    assertEquals(Collections.singletonMap("command", "find"), findFailureLatencyName.tags());
+  }
+
+  @Test
+  public void testSuccessfulCommand() {
```

Comment:
(nit) we don't have to change it for this PR, but stylistically for future tests prefer

1. naming something `shouldXYZ` instead of `testXYZ` since it encourages not only what functionality its testing but also describes the success criteria. we noticed this results in more targeted tests that are easier to debug
2. add comments for Given, When, Then -- this structures tests in a way that's easy to read and consistent throughout the codebase (setup, the test, the assertions)

Reply:
Sounds fair to me.